### PR TITLE
[Android] Speed limit in navigation mode

### PR DESCRIPTION
**Pull request by strump**
Changed navigation UI to show speed and speed limit with `'/'` separator (as in iOS).

### Q1. How to test it without having a ride in a car?

How to check it on emulator?

**Expected result:** speed limit should be displayed in navigation mode:

ios speed limit

### Testing:

Screenshot 2024-08-09 at 10 55 04

I used GPX track [Florida. Busch Gardens - Pollo Tropical.gpx](https://gist.github.com/strump/d4b8cadde7324e4cba5dc802b9d9cb99) to emulate city ride.

### CODE REVIEW

**Comment by biodranik** [comment]
@strump Thanks! Try to google for a gpx file from any car ride in mapped areas (maybe some are available at osm.org) and use it to feed into Android Emulator.
https://stackoverflow.com/questions/72179679/how-can-i-drive-in-android-simulated-phones-when-using-maps

File: android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java [hunk 1]
```diff
@@ -211,8 +216,20 @@ private void updateSpeedView(@NonNull RoutingInfo info)
 
     Pair<String, String> speedAndUnits = StringUtils.nativeFormatSpeedAndUnits(last.getSpeed());
 
+    if (mCurrentSpeedLimit > 0.0)
+    {
+      Pair<String, String> speedLimitAndUnits = StringUtils.nativeFormatSpeedAndUnits(last.getSpeed());
+      mSpeedValue.setText(speedAndUnits.first + " / " + speedLimitAndUnits.first);
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java:222]: Would this narrow non-breaking space look better? https://www.compart.com/en/unicode/U+202F

Comment (biodranik) [reply android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java:222]: It can also be used on iOS btw.

Comment (strump) [reply android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java:222]: You mean replace `" / "` with `"\u202F/\u202F"`?

Comment (biodranik) [reply android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java]: Yes, or just copy the necessary space symbol and insert it directly into the string.

File: android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java [hunk 2]
```diff
@@ -211,9 +214,20 @@ private void updateSpeedView(@NonNull RoutingInfo info)
...
+      mSpeedValue.setText(speedAndUnits.first);
+
+    if (info.isSpeedLimitExceeded())
+      mSpeedValue.setTextColor(ContextCompat.getColor(mActivity, R.color.base_red));
+    else
+      mSpeedValue.setTextColor(ThemeUtils.getColor(mActivity, android.R.attr.textColorPrimary));
+
     mSpeedUnits.setText(speedAndUnits.second);
-    mSpeedValue.setText(speedAndUnits.first);
-    mSpeedViewContainer.setActivated(info.isSpeedLimitExceeded());
```
Comment (strump) [inline android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java:216]: This line made speedometer backgound red. Removed it. Now only text color is red.

File: android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java [hunk 3]
```diff
@@ -211,9 +214,28 @@ private void updateSpeedView(@NonNull RoutingInfo info)
...
+    else
+      mSpeedValue.setText(speedAndUnits.first);
+
+    if (last.getSpeed() > info.speedLimitMps)
+    {
+      if (info.isSpeedCamLimitExceeded())
+        // White text on red background for camera speeding
+        mSpeedValue.setTextColor(ContextCompat.getColor(mActivity, R.color.white_primary));
+      else
+        // Black text for speeding if there's no camera
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java:231]:
```suggestion
        // Red text for speeding if there's no camera
```
Or you can drop these comments, the code is self-explanatory.

File: android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java (hunk 2 above)
Comment (biodranik) [reply android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java:216]: This comment is not relevant anymore, right?

**Review by biodranik — APPROVED** [review]
LGTM with nits.

File: android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java [hunk 4]
```diff
@@ -211,9 +214,28 @@ private void updateSpeedView(@NonNull RoutingInfo info)
...
+      Pair<String, String> speedLimitAndUnits = StringUtils.nativeFormatSpeedAndUnits(info.speedLimitMps);
+      mSpeedValue.setText(speedAndUnits.first + "\u202F/\u202F" + speedLimitAndUnits.first);
+    }
+    else
+      mSpeedValue.setText(speedAndUnits.first);
+
+    if (last.getSpeed() > info.speedLimitMps)
+    {
+      if (info.isSpeedCamLimitExceeded())
+        // White text on red background for camera speeding
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java:228]:
```suggestion
```

File: android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java (hunk 3 above)
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java:231]:
```suggestion
```

**Review by biodranik — APPROVED** [review]
Thanks and let's proceed with the release!

**Pull request merged**